### PR TITLE
feat: check if default resource pools are bound

### PR DESCRIPTION
**Pull request by eecsliu**
## Description
Prevents a cluster from starting up if the specified default resource pools have bindings to workspaces.

## Test Plan
Start a cluster, create a workspace, and bind one of your resource pools to the workspace. Go into the config and set:
```
...
  resource_manager:
      ...
      default_compute_resource_pool: <name of the bound pool>
...
```

Attempt to start the cluster up again and see that it fails. Repeat the same with `default_aux_resource_pool`.
Try this with a different resource pool (and with no default resource pools specified) and make sure it succeeds.

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

File: master/internal/db/postgres_cluster.go [hunk 1]
```diff
@@ -131,3 +134,17 @@ func (db *PgDB) UpdateResourceAllocationAggregation() error {
 	}
 	return nil
 }
+
+// CheckIfRPUnbound checks to make sure both of the specified resource pools are not bound to any
+// workspaces and returns an error if they are.
+func (db *PgDB) CheckIfRPUnbound(defaultAux, defaultCompute string) error {
```
Comment (NicholasBlaskey) [inline master/internal/db/postgres_cluster.go:140]: don't add to PgDB

File: master/internal/core.go [hunk 2]
```diff
@@ -852,6 +864,11 @@ func (m *Master) Run(ctx context.Context) error {
 		return errors.Wrap(err, "could not fetch cluster id from database")
 	}
 
+	err = m.checkIfRMDefaultsAreUnbound(m.config.ResourceManager)
+	if err != nil {
+		return errors.Wrap(err, "could not validate cluster default resource pools")
```
Comment (NicholasBlaskey) [inline master/internal/core.go:869]: nit: use ```fmt.Errorf("messasge: %w", err)```

File: master/internal/core.go [hunk 3]
```diff
@@ -820,6 +820,18 @@ func updateClusterHeartbeat(ctx context.Context, db *db.PgDB) {
...
+func (m *Master) checkIfRMDefaultsAreUnbound(rmConfig *config.ResourceManagerConfig) error {
+	if rmConfig.AgentRM != nil {
+		return m.db.CheckIfRPUnbound(rmConfig.AgentRM.DefaultAuxResourcePool,
+			rmConfig.AgentRM.DefaultComputeResourcePool)
+	}
+	if rmConfig.KubernetesRM != nil {
+		return m.db.CheckIfRPUnbound(rmConfig.KubernetesRM.DefaultAuxResourcePool,
+			rmConfig.KubernetesRM.DefaultComputeResourcePool)
+	}
+	return fmt.Errorf("no Resource Manager found")
```
Comment (NicholasBlaskey) [inline master/internal/core.go:832]: do you have an EE PR ready so we don't break slurm?

File: master/internal/db/postgres_cluster.go [hunk 4]
```diff
@@ -131,3 +134,17 @@ func (db *PgDB) UpdateResourceAllocationAggregation() error {
...
+	exists, err := Bun().NewSelect().Table("rp_workspace_bindings").
+		Where("pool_name IN (?)", bun.In([]string{defaultAux, defaultCompute})).
+		Exists(context.TODO())
+	if err != nil {
+		return err
+	} else if exists {
+		return fmt.Errorf("default resource pools can not be bound to any workspaces")
```
Comment (NicholasBlaskey) [inline master/internal/db/postgres_cluster.go:147]: Can we include the pool name that is the issue and also mention if it is the compute vs aux in the error message?

Stopping master from starting could be confusing.

**Review by NicholasBlaskey — APPROVED** [review]
few minor comments

but LGTM

File: master/internal/db/postgres_cluster.go (hunk 1 above)
Comment (eecsliu) [reply master/internal/db/postgres_cluster.go:140]: thought I needed to put it in there for some reason... will move this into the rp workspace bindings db code

File: master/internal/core.go (hunk 3 above)
Comment (eecsliu) [reply master/internal/core.go:832]: yeah that's in progress, won't merge until I have EE ready as well

Comment (eecsliu) [reply master/internal/core.go:840]: https://github.com/determined-ai/determined-ee/pull/1011

**Pull request merged**